fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Encoding.kt

```diff
@@ -65,23 +66,15 @@ class Encoding : Fuel.RequestConvertible {
         return URL(uri.toASCIIString())
     }
 
-    private fun encodeParameterInUrl(method: Method): Boolean {
-        when (method) {
-            Method.GET, Method.DELETE, Method.HEAD -> return true
-            else -> return false
-        }
+    private fun encodeParameterInUrl(method: Method): Boolean = when (method) {
+        Method.GET, Method.DELETE, Method.HEAD -> true
+        else -> false
     }
 
-    private fun queryFromParameters(params: List<Pair<String, Any?>>?): String {
-        return params?.let {
-            params.filterNot { it.second == null }
-                    .mapTo(mutableListOf()) { "${it.first}=${it.second}" }
-                    .joinToString("&")
-        } ?: ""
-    }
-
-    companion object {
-        private fun defaultHeaders(): MutableMap<String, Any> = mutableMapOf("Accept-Encoding" to "compress;q=0.5, gzip;q=1.0")
-    }
+    private fun queryFromParameters(params: List<Pair<String, Any?>>?): String = params.orEmpty()
+            .filterNot { it.second == null }
+            .mapTo(mutableListOf()) { "${it.first}=${it.second}" }
+            .joinToString("&")
 
+    private fun defaultHeaders(): MutableMap<String, Any> = mutableMapOf("Accept-Encoding" to "compress;q=0.5, gzip;q=1.0")
```

Comment:
Having return as `MutableMap` might not be such a great idea. Can we use just `Map` and let the client convert to `mutable` counterpart themselves?